components/authz-service/storage/v2/memstore/memstore.go

```diff
@@ -243,6 +243,7 @@ func (s *State) GetPolicyChangeNotifier(ctx context.Context) (v2.PolicyChangeNot
 }
 
 func (s *State) CreateRule(_ context.Context, rule *storage.Rule) (*storage.Rule, error) {
+	rule.Status = "applied"
```

Comment:
if we want to match postgres behavior, the status here would be `staged`. only on ApplyRule does everything staged get set to `applied`.

Reply:
Leaving this as-is to not break all the memstore-based tests. Let's nuke them. 😉